### PR TITLE
CCD-3853 run updated FTAs against next test definition release

### PR DESCRIPTION
**Pull request by mattnayler**
### JIRA link (if applicable) ###

   [CCD-3853](https://tools.hmcts.net/jira/browse/CCD-3853) _"Intermittent Data Store FT failures in nightly build pipelines"_

### Change description ###

Run **updated FTAs** against the next `ccd-test-definitions` release, see hmcts/ccd-test-definitions#58.

> see also: hmcts/ccd-definition-store-api#1303

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -20,7 +20,8 @@ def branchesToSync = ['demo', 'ithc', 'perftest']
 
 // Variables to switch pipeline logic and wiring per type of build
 
-def definitionStoreDevelopPr = "PR-575"  // This doesn't change frequently, but when it does, only change this value.
+// TODO: remove TEMPORARY OVERRIDE to test ElasticSearch, reinstate: def definitionStoreDevelopPr = "PR-575"
+def definitionStoreDevelopPr = "PR-1303"  // This doesn't change frequently, but when it does, only change this value.
```
Comment (mattnayler) [inline Jenkinsfile_CNP:24]: This temporary change should be reversed before merging.

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -95,7 +96,8 @@ env.CASE_DOCUMENT_AM_URL = "http://ccd-case-document-am-api-aat.service.core-com
 env.RD_LOCATION_REF_API_BASE_URL = "http://rd-location-ref-api-aat.service.core-compute-aat.internal"
 env.RD_PROFESSIONAL_API_BASE_URL = "http://rd-professional-api-aat.service.core-compute-aat.internal"
 env.BEFTA_RESPONSE_HEADER_CHECK_POLICY="JUST_WARN"
-env.ELASTIC_SEARCH_FTA_ENABLED = "false"
+// TODO: remove TEMPORARY OVERRIDE to test ElasticSearch, reinstate: env.ELASTIC_SEARCH_FTA_ENABLED = "false"
+env.ELASTIC_SEARCH_FTA_ENABLED = "true"
```
Comment (mattnayler) [inline Jenkinsfile_CNP:100]: This temporary change should be reversed before merging.

File: charts/ccd-data-store-api/values.preview.template.yaml [hunk 3]
```diff
@@ -27,7 +27,8 @@ java:
     DATA_STORE_DB_PASSWORD: "{{ .Values.postgresql.auth.password}}"
     DATA_STORE_DB_OPTIONS: "?stringtype=unspecified"
     DATA_STORE_DB_MAX_POOL_SIZE: 10
-    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-575-java/
+    # TODO: remove TEMPORARY OVERRIDE to test ElasticSearch, reinstate: DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-575-java/
+    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-1303-java/
```
Comment (mattnayler) [inline charts/ccd-data-store-api/values.preview.template.yaml:31]: This temporary change should be reversed before merging.

**Review by rajaram-nagarajan — APPROVED** [review]
LGTM

File: build.gradle [hunk 4]
```diff
@@ -277,7 +277,7 @@ dependencies {
     implementation 'org.jooq:jool-java-8:0.9.14'
     implementation 'com.github.hmcts:ccd-case-document-am-client:1.7.1'
 
-    testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.19.4'
+    testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.19.6-CCD-3853'
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply build.gradle]: Resolved in a52f219b44f6ce5c04a11170ddaa24ff95ca948b

File: Jenkinsfile_CNP (hunk 1 above)
Comment (mattnayler) [reply Jenkinsfile_CNP]: Resolved in d8e17adbc5211096b3126fd1571b6f345d8f3f62.

File: charts/ccd-data-store-api/values.preview.template.yaml (hunk 3 above)
Comment (mattnayler) [reply charts/ccd-data-store-api/values.preview.template.yaml]: Resolved in d8e17adbc5211096b3126fd1571b6f345d8f3f62.

File: Jenkinsfile_CNP (hunk 2 above)
Comment (mattnayler) [reply Jenkinsfile_CNP]: Resolved in d8e17adbc5211096b3126fd1571b6f345d8f3f62.

**Review by marianadpereira — APPROVED** [review]
LGTM

**Pull request merged**